zio-http/shared/src/main/scala/zio/http/Handler.scala

```diff
@@ -42,18 +48,23 @@ sealed trait Handler[-R, +Err, -In, +Out] { self =>
     aspect.applyHandler(convert(self))
   }
 
-  def @@[Env1 <: R, Ctx, In1 <: In](aspect: HandlerAspect[Env1, Ctx])(implicit
-    zippable: Zippable.Out[Ctx, Request, In1],
-    res: Out <:< Response,
+  def @@[Env0, Ctx <: R, In1 <: In](aspect: HandlerAspect[Env0, Ctx])(implicit
+    in: Handler.IsRequest[In1],
+    out: Out <:< Response,
+    err: Err <:< Response,
     trace: Trace,
-  ): Handler[Env1, Response, Request, Response] = {
-    def convert(handler: Handler[R, Err, In, Out]): Handler[R, Response, In1, Response] =
-      handler.asInstanceOf[Handler[R, Response, In1, Response]]
+    tag: Tag[Ctx],
+    tag1: Tag[Env0],
+  ): Handler[Env0, Response, Request, Response] =
+    aspect.applyHandlerContext {
+      handler { (ctx: Ctx, req: Request) =>
+        val handler: ZIO[Ctx, Response, Response] = self.asInstanceOf[Handler[Ctx, Response, Request, Response]](req)
+        handler.provideSomeLayer[Env0](ZLayer.environment[Env0] ++ ZLayer.succeed(ctx))
```

Comment:
```suggestion
        handler.provideSomeEnvironment[Env0](_.add[Ctx](ctx))
```

`Ctx` type should be annotated for Scala 2.